kratos/input_output/stl_io.cpp

```diff
@@ -52,6 +80,33 @@ void StlIO::ReadModelPart(ModelPart & rThisModelPart)
         ReadSolid(rThisModelPart);
 }
 
+void StlIO::WriteModelPart(ModelPart & rThisModelPart)
+{    
+    (*mpInputStream) << "solid compensated_skin_part\n";
+
+
+    for (auto & rElem : rThisModelPart.Elements()) {
+        const auto & rGeom = rElem.GetGeometry();
+        const auto & rUnitNormal = rGeom.UnitNormal(rGeom.Center());
+        (*mpInputStream) << "    facet normal " << rUnitNormal[0] << " " << rUnitNormal[1] << " " << rUnitNormal[2] << "\n";
+    
+        (*mpInputStream) << "        outer loop\n";
+
+        for(std::size_t i = 0; i < rGeom.size(); i++ )
+        {
+            auto & rNode = rGeom.GetPoint(i);
+            (*mpInputStream) << "           vertex " << rNode.X() << " " << rNode.Y() << " " << rNode.Z() << "\n";
+        }
```

Comment:
I don't know much about the STL format so I might be wrong, but I'm under the impression that it defines a **triangulation**, so you'd either have to restrict the input `ModelPart`s to ones that only contain triangle elements, or perform the triangulation yourself for any element.